lib/pages/state_notifier_sample/state_notifier_sample_controller.dart

```diff
@@ -1,11 +1,101 @@
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:morning_weakers/core/dummy_data.dart';
+import 'package:morning_weakers/models/models.dart';
 import 'package:morning_weakers/pages/state_notifier_sample/state_notifier_sample_state.dart';
+import 'package:morning_weakers/repositories/group_repository.dart';
+import 'package:morning_weakers/repositories/hackathon_repository.dart';
+import 'package:morning_weakers/repositories/participant_repository.dart';
+import 'package:morning_weakers/repositories/questionnaire_repository.dart';
 import 'package:state_notifier/state_notifier.dart';
 
-class SampleController extends StateNotifier<SampleState> {
-  SampleController() : super(SampleState(user: dummyUser()));
+// TODO: 動作確認用、あとで消す
+class SampleController extends StateNotifier<SampleState> with LocatorMixin {
+  SampleController() : super(const SampleState());
 
-  void handleClick(){
+  HackathonRepository get hackathonRepository => read<HackathonRepository>();
+
+  ParticipantRepository get participantRepository => read<ParticipantRepository>();
+
+  GroupRepository get groupRepository => read<GroupRepository>();
+
+  QuestionnaireRepository get questionnaireRepository => read<QuestionnaireRepository>();
+
+  void handleClick() {
     state = state.copyWith(count: state.count + 1);
   }
+
+  Future<void> createHackathon() async {
+    final FirebaseUser firebaseUser = await FirebaseAuth.instance.currentUser();
+    if (firebaseUser == null) {
+      return;
+    }
+    final Participant participant = dummyParticipant(user: dummyUser(id: firebaseUser.uid), isAdmin: true);
+    final Hackathon hackathon = dummyHackathon(participants: [participant], questionnaire: null);
+    await hackathonRepository.createHackathon(hackathon);
+  }
+
+  Future<void> getHackathon() async {
+    final FirebaseUser firebaseUser = await FirebaseAuth.instance.currentUser();
+    if (firebaseUser == null) {
+      return;
+    }
+    final Hackathon hackathon = await hackathonRepository.getHackathon(hackathonRepository.currentHackathonId);
+    state = state.copyWith(hackathon: hackathon);
+  }
```

Comment:
firebaseUserを取得してnullチェックのところまで
結構多いからなんかメソッドとかでまとめられないか考えたけど思いつかなかった
こんな感じで笑
```suggestion
  Future<void> getHackathon() async {
    final nonNullFirebaseUser = nonNullFirebaseUser()
    final Hackathon hackathon = await hackathonRepository.getHackathon(hackathonRepository.currentHackathonId);
    state = state.copyWith(hackathon: hackathon);
  }
```

Reply:
せやねんなあ、Authとつなげるところで工夫してみるから次PRで！